redisraft.c

```diff
@@ -548,11 +549,14 @@ static int cmdRaftCluster(RedisModuleCtx *ctx, RedisModuleString **argv, int arg
     size_t cmd_len;
     const char *cmd = RedisModule_StringPtrLen(argv[1], &cmd_len);
     if (!strncasecmp(cmd, "INIT", cmd_len)) {
-        if (argc != 2) {
+        if (argc < 2 || argc > 3) {
             RedisModule_WrongArity(ctx);
             return REDISMODULE_OK;
         }
         req = RaftReqInit(ctx, RR_CLUSTER_INIT);
+        if (argc == 3) {
+            req->r.cluster_init.id = RedisModule_CreateStringFromString(ctx, argv[2]);
+        }
```

Comment:
How about defining `cluster_init.id` as char[RAFT_DB_ID_LEN + 1] to avoid dynamic string handling? I didn't try the code but something like this may work : 
```suggestion
        char cluster_id[RAFT_DB_ID_LEN + 1];
        if (argc == 2) {
             RedisModule_GetRandomHexChars(cluster_id, RAFT_DBID_LEN);
             cluster_id[RAFT_DBID_LEN] = '\0';
        } else {
            size_t len;
            char *reqId = RedisModule_StringPtrLen(argv[2], &len);
            if (len != RAFT_DBID_LEN) {
                  RedisModule_ReplyWithError(ctx, "ERR cluster id must be 32 characters");
                  return REDISMODULE_OK;
            }
            memcpy(cluster_id, reqId, RAFT_DBID_LEN);
            cluster_id[RAFT_DBID_LEN] = '\0';
        } 
        
        req = RaftReqInit(ctx, RR_CLUSTER_INIT);
        mempy(req->r.cluster_init.id, cluster_id, RAFT_DBID_LEN + 1);
        
```